log-manager/src/main/java/io/airlift/log/Logging.java

```diff
@@ -276,8 +286,10 @@ public synchronized void configure(LoggingConfiguration config)
             }
         }
 
-        if (!config.isConsoleEnabled()) {
-            disableConsole();
+        boolean isLoggingJsonInConsole = config.getConsoleFormat().equals(Format.JSON);
+
+        if (config.isConsoleEnabled() && isLoggingJsonInConsole) {
+            enableJsonConsole(logAnnotations);
         }
```

Comment:
I think this is the wrong place for this code.  Instead in `configure`, replace:

```java
        if (!config.isConsoleEnabled()) {
            disableConsole();
        }
```

```java
        if (config.isConsoleEnabled()) {
            ROOT.removeHandler(consoleHandler);
            // todo do we want logAnnotations on the console output
            consoleHandler = new OutputStreamHandler(System.err, config.getConsoleFormat().createFormatter(logAnnotations));
            ROOT.addHandler(consoleHandler);
        }
        else {
            disableConsole();
        }
```

Reply:
@mattstep and @electrum Do we want the `logAnnotations` sent to the console output. 

In cases where you are doing console logging piped into a log collector, I could see you want that.  Generally, if you are only using one log style, I think we should add the annotations; otherwise, I likely would not want them on the console.

Reply:
For us we are doing console logging piped into a log collector in trino-gateway, so having an option to include `logAnnotations` is beneficial for us